### PR TITLE
feat(spanner/spannertest): support array literals

### PR DESCRIPTION
**Pull request by dsymonds**
These are relatively easy, and often appear with UNNEST.

### CODE REVIEW

**Review by skuruppu — APPROVED** [review]

File: spanner/spannertest/db_eval.go [hunk 1]
```diff
@@ -774,14 +785,29 @@ func (ec evalContext) colInfo(e spansql.Expr) (colInfo, error) {
 		return colInfo{Type: qp.Type}, nil
 	case spansql.Paren:
 		return ec.colInfo(e.Expr)
+	case spansql.Array:
+		// Assume all element of an array literal have the same type.
+		if len(e) == 0 {
+			// TODO: What does the real Spanner do here?
```
Comment (olavloite) [inline spanner/spannertest/db_eval.go:791]: From https://cloud.google.com/spanner/docs/arrays#using_array_literals

You can write an empty array of a specific type using ARRAY<type>[]. You can also write an untyped empty array using [], in which case Cloud Spanner SQL attempts to infer the array type from the surrounding context. If Cloud Spanner SQL cannot infer a type, the default type ARRAY<INT64> is used.

So this seems to be correct.

File: spanner/spansql/parser.go [hunk 2]
```diff
@@ -2707,12 +2731,12 @@ func (p *parser) parseOnDelete() (OnDelete, *parseError) {
 
 // parseCommaList parses a parenthesized comma-separated list,
```
Comment (olavloite) [inline spanner/spansql/parser.go:2732]:
```suggestion
// parseCommaList parses a comma-separated list enclosed between bra and ket,
```

**Review by olavloite — APPROVED** [review]

File: spanner/spansql/parser.go (hunk 2 above)
Comment (dsymonds) [reply spanner/spansql/parser.go:2732]: done.

File: spanner/spannertest/db_eval.go (hunk 1 above)
Comment (dsymonds) [reply spanner/spannertest/db_eval.go:791]: I wasn't sure about the INT64 part of the type though. The real spanner uses INT64 in some situations (e.g. `SELECT NULL`), but I haven't tested out what it does with, say, `SELECT []`. That's all this covers.

**Pull request merged**